charts/event-handler/templates/deployment.yaml

```diff
@@ -0,0 +1,105 @@
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command:
+            - /usr/local/bin/teleport-event-handler
+            - start
+            - "--config"
+            - "/etc/teleport-event-handler.toml"
+          ports:
+            - name: http
+              containerPort: 80
+              protocol: TCP
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          volumeMounts:
+            - name: config
+              mountPath: /etc/teleport-event-handler.toml
+              subPath: teleport-event-handler.toml
+            - name: auth-id
+              mountPath: /var/lib/teleport/plugins/event-handler/auth_id
+              subPath: {{ .Values.teleport.identitySecretPath }}
+            - name: certificate
+              mountPath: /var/lib/teleport/plugins/event-handler/ca.crt
+              subPath: {{ .Values.fluentd.certificate.caPath }}
+            - name: certificate
+              mountPath: /var/lib/teleport/plugins/event-handler/client.crt
+              subPath: {{ .Values.fluentd.certificate.certPath }}
+            - name: certificate
+              mountPath: /var/lib/teleport/plugins/event-handler/client.key
+              subPath: {{ .Values.fluentd.certificate.keyPath }}
+            {{- if or .Values.persistentVolumeClaim.enabled .Values.persistentVolumeClaim.existingClaim }}
+            - name: storage
+              mountPath: "{{ .Values.eventHandler.storagePath }}"
+            {{- end }}
+            {{- with .Values.volumeMounts -}}
+              {{- toYaml . | nindent 12 }}
+            {{- end}}
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      volumes:
+        - name: config
+          configMap:
+            name: {{ include "event-handler.fullname" . }}
+            defaultMode: 0600
+        - name: auth-id
+          secret:
+            secretName: "{{ .Values.teleport.identitySecretName }}"
+            defaultMode: 0600
+        - name: certificate
+          secret:
+            secretName: "{{ .Values.fluentd.certificate.secretName }}"
+            defaultMode: 0600
+        {{- if .Values.persistentVolumeClaim.enabled }}
+        - name: storage
```

Comment:
Since we support feeding in arbitrary volumeMounts, this name should be configurable to make it possible to avoid name conflicts